experimental/parser/legalize_decl.go

```diff
@@ -133,10 +144,22 @@ func legalizeRange(p *parser, parent classified, decl ast.DeclRange) {
 				isName = true
 
 				if m := p.Mode(); m == taxa.EditionMode {
-					p.Errorf("cannot use %vs in %v in %v", taxa.String, in, m).Apply(
+					err := p.Errorf("cannot use %vs in %v in %v", taxa.String, in, m).Apply(
 						report.Snippet(expr),
 						report.Snippetf(p.syntax, "%v is specified here", m),
 					)
+
+					// Only suggest unquoting if it's already an identifier.
```

Comment:
If it's not an identifier, we should report that as an error. FWIW, the current (non-experimental) compiler already considers this an error. (Though it's merely a warning in `protoc`...).

I put this in the language spec regarding this:
> The contents of the string literal that defines the reserved name must represent a valid identifier. In other words, they must match the production for the [_identifier_](https://protobuf.com/docs/language-spec#identifiers-and-keywords) lexical element.

Even though `protoc` does not emit a warning, their published grammar also implies that this is an error: https://protobuf.dev/reference/protobuf/proto2-spec/#reserved

Reply:
That's already an error, it's diagnozed a few stanzas down.